""" known error is no longer a solvable

Revision ID: 3b8bedc278b6
Revises: 681ad0738ddc
Create Date: 2022-06-29 16:13:43.802937

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = '3b8bedc278b6'
down_revision = '681ad0738ddc'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_constraint('known_error_created_by_fkey', 'known_error', type_='foreignkey')
    op.drop_constraint('known_error_taken_by_fkey', 'known_error', type_='foreignkey')
    op.drop_column('known_error', 'is_blocked')
    op.drop_column('known_error', 'taken_by')
    op.drop_column('known_error', 'status')
    op.drop_column('known_error', 'priority')
    op.drop_column('known_error', 'created_by')
    op.drop_column('known_error', 'description')
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.add_column('known_error', sa.Column('description', sa.VARCHAR(length=500), autoincrement=False, nullable=True))
    op.add_column('known_error', sa.Column('created_by', sa.VARCHAR(), autoincrement=False, nullable=True))
    op.add_column('known_error', sa.Column('priority', sa.VARCHAR(length=20), autoincrement=False, nullable=True))
    op.add_column('known_error', sa.Column('status', sa.VARCHAR(length=20), autoincrement=False, nullable=True))
    op.add_column('known_error', sa.Column('taken_by', sa.VARCHAR(), autoincrement=False, nullable=True))
    op.add_column('known_error', sa.Column('is_blocked', sa.BOOLEAN(), autoincrement=False, nullable=True))
    op.create_foreign_key('known_error_taken_by_fkey', 'known_error', 'user', ['taken_by'], ['username'])
    op.create_foreign_key('known_error_created_by_fkey', 'known_error', 'user', ['created_by'], ['username'])
    # ### end Alembic commands ###
